src/lib/snack-bar/snack-bar.spec.ts

```diff
@@ -259,6 +259,32 @@ describe('MdSnackBar', () => {
     // Let remaining animations run.
     tick(500);
   }));
+
+  it('should dismiss the snackbar when the action is called, notifying of both action and dismiss',
+     fakeAsync(() => {
+       let dismissObservableCompleted = false;
+       let actionObservableCompleted = false;
+       let snackBarRef = snackBar.open('Some content', 'dismiss');
+       viewContainerFixture.detectChanges();
+       
+       snackBarRef.afterDismissed().subscribe(null, null, () => {
+         dismissObservableCompleted = true;
+       });
+       snackBarRef.onAction().subscribe(null, null, () => {
+         actionObservableCompleted = true;
+      });
+
+      snackBarRef.action();
```

Comment:
Instead of calling this directly, could you grab the action button element and call `.click()` on it?